File: spec/journey/find_non_domestic_certificate_by_postcode_spec.rb
# frozen_string_literal: true

describe 'Find Non-Dom EPC By Postcode in English', journey: true do
  context 'when searching for a domestic certificate' do
    before do
      visit find_service
      click_on 'Start now'
      find('#label-non-domestic').click
      click_on 'Continue'
      fill_in 'postcode', with: 'SW1A 2AA'
      click_button 'Find'
      click_on 'DEC', match: :first
    end

    it 'shows the certificate with the expected header' do
      expect(page).to have_content 'Display energy certificate (DEC)'
    end
  end
end

describe 'Find Non-Dom EPC By Postcode in Welsh', journey: true do
  context 'when searching for a domestic certificate in Welsh' do
    before do
      visit find_service_welsh
      click_on 'Dechrau nawr'
      find('#label-non-domestic').click
      click_on 'Parhau'
      fill_in 'postcode', with: 'SW1A 2AA'
      click_button 'Chwiliwch'
      click_on 'DEC', match: :first
    end

    it 'shows the certificate with the expected header' do
      expect(page).to have_content 'Tystysgrif ynni i’w harddangos (DEC)'
    end
  end
end

File: cypress/integration/find_non_domestic_certificate_by_postcode_spec.js
describe('Find non-domestic EPC by postcode (English)', () => {
  beforeEach(() => {
    cy.visit(Cypress.env(`find_service_en_${process.env.API_STAGE || 'integration'}`))
    cy.contains('Start now').click()
    cy.get('#label-non-domestic').click()
    cy.contains('Continue').click()
    cy.get('input[name=postcode]').type('SW1A 2AA')
    cy.contains('button', 'Find').click()
    cy.contains('DEC').click()
  })

  it('shows the certificate with the expected header', () => {
    cy.get('body').should('contain', 'Display energy certificate (DEC)')
  })
})

describe('Find non-domestic EPC by postcode (Welsh)', () => {
  beforeEach(() => {
    cy.visit(Cypress.env(`find_service_cy_${process.env.API_STAGE || 'integration'}`))
    cy.contains('Dechreuwch nawr').click()
    cy.get('#label-non-domestic').click()
    cy.contains('Parhau').click()
    cy.get('input[name=postcode]').type('SW1A 2AA')
    cy.contains('button', 'Chwiliwch').click()
    cy.contains('DEC').click()
  })

  it('shows the certificate with the expected header', () => {
    cy.get('body').should('contain', 'Tystysgrif ynni i’w harddangos (DEC)')
  })
})
